From 8d90ebfdb06f977d001e527a948f7477186776f4 Mon Sep 17 00:00:00 2001
From: Jayanth Othayoth <ojayanth@gmail.com>
Date: Wed, 22 Jul 2026 03:23:00 -0500
Subject: [PATCH 04/33] test:openpower-pels: sync extension build settings

The openpower-pels tests compile sources from
log_manager_ext_sources, but do not inherit all extension build
settings used by the main log manager build.

As extensions evolve, they may require additional dependencies or
include directories. This can cause test targets to fail to build
while the production log manager build continues to succeed.

Propagate extension dependencies and include directories to the
test targets so extension sources are built consistently in both
environments

Change-Id: I4763aa0f410ffbe5ac398060948b5b624c40e348
Signed-off-by: Jayanth Othayoth <ojayanth@gmail.com>
---
 meson.build                     | 9 +++++++--
 test/openpower-pels/meson.build | 8 ++++++--
 2 files changed, 13 insertions(+), 4 deletions(-)

diff --git a/meson.build b/meson.build
index 857dcd4..95ec943 100644
--- a/meson.build
+++ b/meson.build
@@ -149,6 +149,7 @@ elog_process_gen = custom_target(
 log_manager_ext_sources = []
 log_manager_ext_deps = []
 log_manager_ext_args = []
+log_manager_ext_inc = []
 
 subdir('extensions')
 subdir('phosphor-rsyslog-config')
@@ -184,14 +185,18 @@ log_manager_lib = static_library(
     'log_manager_lib',
     log_manager_sources,
     log_manager_ext_sources,
-    include_directories: include_directories('gen'),
+    include_directories: [
+        include_directories('gen'),
+    ] + log_manager_ext_inc,
     dependencies: [log_manager_deps, log_manager_ext_deps],
     cpp_args: log_manager_ext_args,
 )
 executable(
     'phosphor-log-manager',
     'log_manager_main.cpp',
-    include_directories: include_directories('gen'),
+    include_directories: [
+        include_directories('gen'),
+    ] + log_manager_ext_inc,
     cpp_args: log_manager_ext_args,
     dependencies: [conf_h_dep, log_manager_deps, phosphor_logging_dep],
     install: true,
diff --git a/test/openpower-pels/meson.build b/test/openpower-pels/meson.build
index 5528522..a1fb868 100644
--- a/test/openpower-pels/meson.build
+++ b/test/openpower-pels/meson.build
@@ -69,7 +69,9 @@ openpower_test_lib = static_library(
     libpel_sources,
     peltool_sources,
     '../common.cpp',
-    include_directories: include_directories('../../', '../../gen'),
+    include_directories: [
+        include_directories('../../', '../../gen'),
+    ] + log_manager_ext_inc,
     dependencies: [libpel_deps, peltool_deps, gtest_dep],
 )
 
@@ -92,7 +94,9 @@ foreach t : openpower_pels.keys()
                 log_manager_ext_deps,
                 openpower_pels.get(t).get('deps', []),
             ],
-            include_directories: include_directories('../../', '../../gen'),
+            include_directories: [
+                include_directories('../../', '../../gen'),
+            ] + log_manager_ext_inc,
         ),
     )
 endforeach
-- 
2.54.0

